broadcast/src/buffered/engine.rs

```diff
@@ -337,7 +341,17 @@ impl<
 
     /// Respond to a waiter with a message.
     /// Increments the appropriate metric based on the result.
-    fn respond(&mut self, responder: oneshot::Sender<M>, msg: M) {
+    fn respond_subscribe(&mut self, responder: oneshot::Sender<M>, msg: M) {
+        let result = responder.send(msg);
+        self.metrics.subscribe.inc(match result {
+            Ok(_) => Status::Success,
+            Err(_) => Status::Dropped,
+        });
+    }
+
+    /// Respond to a waiter with an optional message.
+    /// Increments the appropriate metric based on the result.
+    fn respond_get(&mut self, responder: oneshot::Sender<Option<M>>, msg: Option<M>) {
```

Comment:
There should probably be three statuses here (check if `msg.is_none`), include `Failure` as one of the three cases. It's hard for me to say definitively the best distinction between `Dropped` and `Failure` here, but I'm currently thinking `Failure` should be if `msg` is `None` but `result` is `Ok`

Reply:
Took a stab at this. I think the right way to think about it is if the listener gets our result (in `get` they always will pretty much) it is `Success/Failure`, if they don't it is just `Dropped` (because it doesn't matter what we did anyways).